core/src/main/java/tech/tablesaw/api/DoubleColumn.java

```diff
@@ -570,10 +570,18 @@ public DoubleColumn subtract(double value) {
         return result;
     }
 
-    public DoubleColumn subtract(DoubleColumn column2) {
-        DoubleColumn result = new DoubleColumn(name() + " - " + column2.name(), size());
-        for (int r = 0; r < size(); r++) {
-            result.append(subtract(get(r), column2.get(r)));
+    public NumericColumn subtract(NumericColumn column2) {
+        return DoubleColumn.subtractDouble(this, column2);
+    }
+
+    public static DoubleColumn subtractDouble(NumericColumn column1, NumericColumn column2) {
+        int col1Size = column1.size();
+        int col2Size = column2.size();
+        int resultSize = col1Size < col2Size ? col1Size : col2Size;
```

Comment:
There's a lot of different ways I can imagine this working if the columns are not the same size. E.g. you could take the larger column and fill in missing values at the end. Is there any reason you chose this behavior? I'm wondering if we should just `throw new IllegalArgumentException("Columns have unequal size " + column1.size() + " and " + column2.size())`

Reply:
I don't have a strong reasoning for this choice. Some tools (for e.g. pandas) do retain the size of the larger column and fill missing values. The api such tools let the user provide a optional "fill" value. It is not easy to add optional parameters in java without making the api cumbersome. So I chose to save memory and ops by retaining the smaller column size. Instead of adding optional parameters I prefer throwing error as it forces the user to think (like Optionals). Let me know if you think throwing error matches rest of the implementation.

Reply:
I think throwing an error would be fine for now. I'd rather start out supporting less and then see where the need arises before introducing additional complexity

Reply:
for the record, i made it throw an exception